src/raven.js

```diff
@@ -843,6 +843,38 @@ function makeRequest(opts) {
     img.src = src;
 }
 
+function makeXhrRequest(opts) {
+    var request = new XMLHttpRequest();
+    request.onreadystatechange = function (e) {
+        if (request.readyState !== 4) {
+            return;
+        }
+
+        if (request.status === 200) {
+            if (opts.onSuccess) {
+                opts.onSuccess();
+            }
+        } else {
+            if (opts.onError) {
+                opts.onError();
+            }
+        }
+    };
+
+    // NOTE: auth is intentionally sent as part of query string (NOT as custom
+    //       HTTP header) so as to avoid preflight CORS requests
+    request.open('POST', opts.url + '?' + urlencode(opts.auth));
+    request.send(JSON.stringify(opts.data));
+}
+
+function makeRequest(opts) {
+    var hasCORS =
+        'withCredentials' in new XMLHttpRequest() ||
+        typeof XDomainRequest !== 'undefined';
+
+    return (hasCORS ? makeXhrRequest : makeImageRequest)(opts);
```

Comment:
I'm just going to assume that this is going to be problematic to fall back to the image request, because the goal of XHR is to be able to transmit *more* data. And if we attempt to fall back, it's unlikely things are even going to work.

So I guess we need to make a decision here: do we just 100% remove the image transport, and assume if there's no CORS support, maybe we just ignore entirely?

/cc @dcramer